src/main/java/com/commercetools/sync/services/CustomerService.java

```diff
@@ -0,0 +1,123 @@
+package com.commercetools.sync.services;
+
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.customers.Customer;
+import io.sphere.sdk.customers.CustomerDraft;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+public interface CustomerService {
+
+    /**
+     * Given a set of keys this method caches in-memory a mapping of key -&gt; id only for those keys which are not
+     * already cached.
+     *
+     * @param keysToCache a set of keys to cache.
+     * @return a map of key to ids of the requested keys.
+     */
+    @Nonnull
+    CompletionStage<Map<String, String>> cacheKeysToIds(@Nonnull Set<String> keysToCache);
+
+    /**
+     * Given a {@link Set} of customer keys, this method fetches a set of all the customers, matching the given set of
+     * keys in the CTP project, defined in an injected {@link SphereClient}. A mapping of the key to the id of the
+     * fetched customers is persisted in an in-memory map.
+     *
+     * @param customerKeys set of customer keys to fetch matching resources by.
+     * @return {@link CompletionStage}&lt;{@link Set}&lt;{@code U}&gt;&gt; in which the result of it's completion
```

Comment:
Actually I don't know what is {@code U} could be a copy-paste issue from the old docs 🤔